### ISSUE TITLE
Prob returned by fit_tranform

### ISSUE THREAD

**Issue by peiguo9028** [MaartenGr/BERTopic#1508]
I ran the fit_tranform() but the returned array prob did not add up to 1. For a data point assigned to a non -1 cluster, I assume the prob should add up to 1. I checked the source code and it seems the probability calculation is from hdbscan package. If the prob represents the chance of a data belonging to a cluster, the total probability should be 1. If not, can you explain what those values represent. In my case, a big portion of prob sum to a value < 0.5.

**Comment by MaartenGr** [comment]
That is correct. The probabilities all represent non-outlier topics. This means that the probability of the outlier topic is 1 minus the sum of all other probabilities.

**Comment by peiguo9028** [comment]
> That is correct. The probabilities all represent non-outlier topics. This means that the probability of the outlier topic is 1 minus the sum of all other probabilities.

Thanks for your reply. In the case that a data point assigned to a non -1 cluster with a very low probability, for example, 0.2, why it’s not marked as an outlier? Is that the case of an in-cluster outlier according to the hdbscan documentation about outlier detection?

**Comment by MaartenGr** [comment]
That might indeed be the case. Also, if 0.2 is the highest value amongst dozens or even hundreds of topics, it is still a relatively high probability.

**Comment by peiguo9028** [comment]
> That might indeed be the case. Also, if 0.2 is the highest value amongst dozens or even hundreds of topics, it is still a relatively high probability.

In my case, only 7 topics, and all probability add up to <.4. But the portion of this kinda low prob instances within a topic is relatively small, not negligibly small though.

**Comment by MaartenGr** [comment]
The thing is, the probabilities generated with HDBSCAN are actually not the probabilities that HDBSCAN uses to assign documents to clusters. It is merely an approximation of the probabilities after fitting the cluster model. Hence, they will not exactly represent the topic assignment.
